src/types/utils.cpp

```diff
@@ -377,6 +1163,19 @@ til::color Utils::ColorFromHexString(const std::string_view str)
     return til::color{ r, g, b };
 }
 
+// Function Description:
+// - Parses a color from a string based on the XOrg app color name table.
+// Arguments:
+// - str: a string representation of the color name to parse
+// Return Value:
+// - A til::color if the string could successfully be parsed. If the string is not
+//      in the table, throws runtime_error
+til::color Utils::ColorFromXOrgAppColorName(const std::wstring_view wstr)

```

Comment:
According to the way VTE implements this, the name is actually case-insensitve, making for example `mArooN` the same as `maroon`. However `static_map` does not (or will IMO) offer this kind of ability. For now I think it's good enough. Besides I highly doubt that people would use something as weird as `mArooN`.

Reply:
If we use a custom comparator instead of std::less, we can get case insensitivity

Reply:
actually, we should just store all the names using the ordinal lowercase values and make this function check against the ordinal lowercase value.

Reply:
For the record, every terminal that I've tested handles the color names case-insensitively (at least of those that support names). So I think it would be preferable if we matched that.

Reply:
So I did it, making everything lowercase. And now spell checking is giving me a hard time..